### ISSUE TITLE
Warnings-as-errors need cleaned up

### ISSUE THREAD

**Issue by prwolfe** [trilinos/Trilinos#2281]
All three builds that use watnings-as-errors are showing 50+ failures.  We need to look into these to support customers that use this. See https://testing.sandia.gov/cdash/index.php?project=Trilinos&parentid=3400360

@trilinos/Trilinos Couplings 3 errors from 2 files
     (packages/trilinoscouplings/src/ml/NonlinML/nlnml_preconditioner_utils.cpp) has unused varliables
     (packages/trilinoscouplings/src/ml/NonlinML/nlnml_preconditioner1.cpp) due to initialization order.

## Expectations

These warnings/errors should be resolved and the changes set as pull requests into develop to support Sierra and other codes. When those pull requests are merged the ticket should be closed.

## Current Behavior

Warnings-as-errors builds fail on the above conditions

## Your Environment

- **Relevant repo SHA1s:**  
- **Relevant configure flags or configure script:**  
- **Operating system and version:**  
- **Compiler and TPL versions:**  

## Related Issues

* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of

**Comment by jmgate** [comment]
Looks like @trilinos/trilinos-couplings isn't a team.  @trilinos/framework, any idea who the right contact here is?

**Comment by jwillenbring** [comment]
@prwolfe Is Trilinos-couplings enabled for customer builds? If not, we should disable this package for the warnings-as-errors builds, if so, is this something  you can look at? The latest breakdown I see assigns Trilinos-couplings to the Framework product.

**Comment by mhoemmen** [comment]
See #2293.  I couldn't figure out how to make the code actually _build_; see notes on the PR.

**Comment by mhoemmen** [comment]
I did get the code actually to build (see #2293).  I'm testing now.

**Comment by mhoemmen** [comment]
Tests pass for me w/ ML, Epetra, and TrilinosCouplings enabled.  I merged into develop.  Thanks @prwolfe !

**Comment by prwolfe** [comment]
Looks like there are still a number of these (all different from the original set ;-).  I will look at them today.

**Comment by prwolfe** [comment]
@william76 - Those changes do not fix the example code.  I just put in a pull request to fix those as well.

**Comment by prwolfe** [comment]
Pull request testing came up clean - closing.
